### PR TITLE
Lecture 9 homework

### PR DESCRIPTION
**Pull request by agrren**

### CODE REVIEW

File: Reneiski_Homework1/Reneiski_Homework1/Program.cs [hunk 1]
```diff
@@ -1,28 +1,279 @@
...
-            Belaz.SetWorker(semen);
-
-            Belaz.ShowAllEmployeeInfo();
-            Belaz.ShowEmployeeNameInfo();
-            Belaz.ShowEmployeeAgeInfo();
-            Belaz.ShowEmployeePositionInfo();
-            Belaz.ShowEmployeeNumber();
-            Belaz.ShowEmployeeByPosition();
-            Belaz.ShowEmployeeWork();
+            //Lists for Tasks
```
Comment (ashurakova) [inline Reneiski_Homework1/Reneiski_Homework1/Program.cs]: could you please move all methods that contains tasks logic into separate class with static methods? and in main just call these static methods?

File: Reneiski_Homework1/Reneiski_Homework1/Program.cs [hunk 2]
```diff
@@ -1,28 +1,279 @@
...
+                    twoNode = twoNode.Next;
+                }
+            }
+
+            //Объединить два связанных списка чисел, включив в окончательный список только те числа, которые встречаются в обоих списка.
+            //Итак, если списки [1,3,4,7,12] и [1,5,7,9], результат будет [1,7].
+
+            static void UniteTwoLinkedLists()
+            {
+                List<string> intsOne = new List<string>() { "1", "3", "4", "7", "12" };
```
Comment (ashurakova) [inline Reneiski_Homework1/Reneiski_Homework1/Program.cs]: use var here and everywhere

File: Reneiski_Homework1/Reneiski_Homework1/Program.cs [hunk 3]
```diff
@@ -1,28 +1,279 @@
...
+                    Console.WriteLine(item);
+                }
+            }
+
+            //Создайте LinkedList и два элемента, вставьте второй элемент после каждого вхождения первого элемента в списке.
+            //Итак, если список равен [2,4,3,2,8,2,5,1,2], а элементы равны 2 и 10, результатом будет [2,10,4,3,2,10,8,2,10,5,1,2,10]
+
+            static void LinkedListAddTen()
+            {
+                List<string> ints = new List<string>() { "2", "4", "3", "2", "8", "2", "5", "1", "2" };
```
Comment (ashurakova) [inline Reneiski_Homework1/Reneiski_Homework1/Program.cs]: I think it's better to create List wih ints

File: Reneiski_Homework1/Reneiski_Homework1/Program.cs [hunk 4]
```diff
@@ -1,28 +1,279 @@
...
+
+                var linkedInts = new LinkedList<string>(ints);
```
Comment (ashurakova) [inline Reneiski_Homework1/Reneiski_Homework1/Program.cs]: you may create LinkedList from start with all elements above (no need to use List separately)

File: Reneiski_Homework1/Reneiski_Homework1/Program.cs [hunk 5]
```diff
@@ -1,28 +1,279 @@
...
+                }
+            }
+
+            //Объединить два связанных списка чисел, включив в окончательный список только те числа, которые встречаются в обоих списка.
+            //Итак, если списки [1,3,4,7,12] и [1,5,7,9], результат будет [1,7].
+
+            static void UniteTwoLinkedLists()
+            {
+                List<string> intsOne = new List<string>() { "1", "3", "4", "7", "12" };
+                List<string> intsTwo = new List<string>() { "1", "5", "7", "9" };
```
Comment (ashurakova) [inline Reneiski_Homework1/Reneiski_Homework1/Program.cs]: please use int here and below/above

File: Reneiski_Homework1/Reneiski_Homework1/Program.cs [hunk 6]
```diff
@@ -1,28 +1,279 @@
...
+            }
+
+            //Создайте Queue. Добавьте занчения в очередь с при помощи ввода с консоли.
+            //Реализуйте операцию: GetMaxValue, которая возвращает максимальное значение, хранящееся в данный момент в очереди, не удаляя его.
+            //Каждый раз, когда текущий максимум удаляется из очереди, операция GetMaxValue очереди должна отражать новый самый большой сохраненный элемент.
+
+            var chisla = new List<int>() { 99, 15, 3 };
+            Queue<int> numbers = new Queue<int>(chisla);
+
+            //AddValueToQueue(ref numbers);
```
Comment (ashurakova) [inline Reneiski_Homework1/Reneiski_Homework1/Program.cs]: should it be commented?

File: Reneiski_Homework1/Reneiski_Homework1/Program.cs [hunk 7]
```diff
@@ -1,28 +1,279 @@
...
+                {
+                    Console.WriteLine(item);
+                }
+            }
+
+            //Создайте Queue. Добавьте занчения в очередь с при помощи ввода с консоли.
+            //Реализуйте операцию: GetMaxValue, которая возвращает максимальное значение, хранящееся в данный момент в очереди, не удаляя его.
+            //Каждый раз, когда текущий максимум удаляется из очереди, операция GetMaxValue очереди должна отражать новый самый большой сохраненный элемент.
+
+            var chisla = new List<int>() { 99, 15, 3 };
```
Comment (ashurakova) [inline Reneiski_Homework1/Reneiski_Homework1/Program.cs]: I guess you need to get all values for queue from console

File: Reneiski_Homework1/Reneiski_Homework1/Program.cs [hunk 8]
```diff
@@ -1,28 +1,279 @@
...
+            //Напишите программу, которая принимает на вход три буквы и отображает их в обратном порядке. Используйте Stack.
+
+            //StackAction();
+
+            static void StackAction()
+            {
+                var bukva = new List<string>() { "a", "b", "c" };
+                Stack<string> letters = new Stack<string>(bukva);
+
+                foreach (Object item in letters)
```
Comment (ashurakova) [inline Reneiski_Homework1/Reneiski_Homework1/Program.cs]: use var

File: Reneiski_Homework1/Reneiski_Homework1/Program.cs [hunk 9]
```diff
@@ -1,28 +1,279 @@
...
+                    Console.WriteLine($"{item}, {cities[item].Population}, {cities[item].Square}");
+                }
+            }
+
+            Console.WriteLine();
+            Console.WriteLine("Sort dictionary by Square");
+
+            var sortedBySquare = cities.OrderBy(p => p.Value.Square);
+
+            var printSortedBySquare = sortedBySquare.ToDictionary(p => p.Key, p => p.Value);
```
Comment (ashurakova) [inline Reneiski_Homework1/Reneiski_Homework1/Program.cs]: I think you can remove this variable and just apply ToDictionary method to cities.OrderBy(p => p.Value.Square) expression

File: Reneiski_Homework1/Reneiski_Homework1/Program.cs [hunk 10]
```diff
@@ -1,28 +1,279 @@
...
+
+            //PrintCitiesDict(printSortedBySquare);
+
+            Console.WriteLine();
+            Console.WriteLine("Sort dictionary by Population");
+
+            var sortedByPopulation = cities.OrderByDescending(p => p.Value.Population);
+
+            var printsortedByPopulation = sortedByPopulation.ToDictionary(p => p.Key, p => p.Value);
```
Comment (ashurakova) [inline Reneiski_Homework1/Reneiski_Homework1/Program.cs]: the same as with printSortedBySquare

File: Reneiski_Homework1/Reneiski_Homework1/Program.cs [hunk 11]
```diff
@@ -1,28 +1,279 @@
...
+
+            //PrintCitiesDict(printsortedByPopulation);
+
+            Console.WriteLine();
+            Console.WriteLine("Total Sum of Population");
+
+            var sumOfPopulation = cities.Sum(p => p.Value.Population).ToString();
```
Comment (ashurakova) [inline Reneiski_Homework1/Reneiski_Homework1/Program.cs]: may be no need to cast it to string? sound like numeric variable

File: Reneiski_Homework1/Reneiski_Homework1/Program.cs (hunk 1 above)
Comment (agrren) [reply Reneiski_Homework1/Reneiski_Homework1/Program.cs]: Fixed

File: Reneiski_Homework1/Reneiski_Homework1/Program.cs (hunk 2 above)
Comment (agrren) [reply Reneiski_Homework1/Reneiski_Homework1/Program.cs]: Fixed

File: Reneiski_Homework1/Reneiski_Homework1/Program.cs (hunk 3 above)
Comment (agrren) [reply Reneiski_Homework1/Reneiski_Homework1/Program.cs]: Fixed

File: Reneiski_Homework1/Reneiski_Homework1/Program.cs (hunk 4 above)
Comment (agrren) [reply Reneiski_Homework1/Reneiski_Homework1/Program.cs]: Fixed

File: Reneiski_Homework1/Reneiski_Homework1/Program.cs (hunk 5 above)
Comment (agrren) [reply Reneiski_Homework1/Reneiski_Homework1/Program.cs]: Fixed

File: Reneiski_Homework1/Reneiski_Homework1/Program.cs (hunk 6 above)
Comment (agrren) [reply Reneiski_Homework1/Reneiski_Homework1/Program.cs]: Nope! Fixed!

File: Reneiski_Homework1/Reneiski_Homework1/Program.cs (hunk 7 above)
Comment (agrren) [reply Reneiski_Homework1/Reneiski_Homework1/Program.cs]: Fixed

File: Reneiski_Homework1/Reneiski_Homework1/Program.cs (hunk 8 above)
Comment (agrren) [reply Reneiski_Homework1/Reneiski_Homework1/Program.cs]: Fixed

File: Reneiski_Homework1/Reneiski_Homework1/Program.cs (hunk 9 above)
Comment (agrren) [reply Reneiski_Homework1/Reneiski_Homework1/Program.cs]: Fixed

File: Reneiski_Homework1/Reneiski_Homework1/Program.cs (hunk 10 above)
Comment (agrren) [reply Reneiski_Homework1/Reneiski_Homework1/Program.cs]: Fixed

File: Reneiski_Homework1/Reneiski_Homework1/Program.cs (hunk 11 above)
Comment (agrren) [reply Reneiski_Homework1/Reneiski_Homework1/Program.cs]: Yeap! Fixed!

File: Reneiski_Homework1/Reneiski_Homework1/Program.cs [hunk 12]
```diff
@@ -1,28 +1,12 @@
-using Reneiski_Homework1.Lecture5;
-using WorkerNamespace;
+﻿using Reneiski_Homework1.Lecture9;
 
 namespace Reneiski_Homework1
 {
-    public class Program
+    public class Program : Lecture9Task
```
Comment (ashurakova) [inline Reneiski_Homework1/Reneiski_Homework1/Program.cs]: not a good idea to inherit from Lecture9Task
as you have static methods in Lecture9Task.cs, you can just write Lecture9Task.Task2() in your Main method

Comment (agrren) [reply Reneiski_Homework1/Reneiski_Homework1/Program.cs]: Fixed

**Pull request merged**